smack-extensions/src/main/java/org/jivesoftware/smackx/softwareinfo/element/SoftwareInfoElement.java

```diff
@@ -0,0 +1,212 @@
+/**
+ *
+ * Copyright 2019 Aditya Borikar
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jivesoftware.smackx.softwareinfo.element;
+
+import java.util.List;
+
+import org.jivesoftware.smack.packet.ExtensionElement;
+import org.jivesoftware.smack.packet.XmlEnvironment;
+import org.jivesoftware.smack.util.EqualsUtil;
+import org.jivesoftware.smack.util.HashCode;
+import org.jivesoftware.smack.util.XmlStringBuilder;
+
+import org.jivesoftware.smackx.mediaelement.element.MediaElement;
+import org.jivesoftware.smackx.xdata.FormField;
+import org.jivesoftware.smackx.xdata.packet.DataForm;
+import org.jivesoftware.smackx.xdata.packet.DataForm.Type;
+
+/**
+ * {@link ExtensionElement} that contains Software Information. <br>
+ * Instance of SoftwareInfoElement can be created using {@link Builder#build()} method.
+ * To obtain Software Information use {@link #getForm()}.getFields()
+ */
+public final class SoftwareInfoElement implements ExtensionElement {
```

Comment:
You do not want to subclass extension element. XEP-0232 is based on data forms. I think you want to mimic the pattern pubsub's `ConfigureForm` and `SubscribeForm` are doing.